oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/session/OidcProviderSessionRegistrationDetails.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.authentication.session;
+
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.web.csrf.CsrfToken;
+
+/**
+ * A registration of the OIDC Provider Session with the Client's session
+ *
+ * @author Josh Cummings
+ * @since 6.1
+ */
+public interface OidcProviderSessionRegistrationDetails {
```

Comment:
Why do we need an interface? Do you think we'll need multiple implementations? Furthermore, the oauth2 functionality doesn't make use of `*Details` naming so we should consider aligning with current naming conventions.

Reply:
The reason is the same as `UserDetails`, so that folks who are mapping to a database don't need to map to the concrete type, but instead can use the bridge pattern we often talk about in that case with `UserDetailsService`. Clearly, we don't have to do this; but it would be hard to change return types after going GA, which is why I'm including it now.

I'm open to changing it as Spring Security does not often code to interfaces when it comes to domain objects. For now, I'll change it to the concrete type.